src/app.rs

```diff
@@ -1,110 +1,148 @@
+        .arg(arg("pattern"))
+        .arg(arg("path"))
+}
+
+#[cfg_attr(rustfmt, rustfmt_skip)]
+fn usage() -> HashMap<&'static str, Help> {
+    let mut h = HashMap::new();
+    doc!(h, "hidden"
+        , "Search hidden files and directories"
+        , "Includes, in a search query, any hidden files and directories for the directory of \
+            the search.");
+    doc!(h, "no-ignore"
+        , "Do not respect .(git)ignore files"
+        , "Searches for a pattern match with files and directories, including all defined \
+            `.*ignore` file exceptions. Useful when you want to match inclusively for a pattern.");
+    doc!(h, "case-sensitive"
+        , "Case-sensitive search (default: smart case)"
+        , "Define your pattern as case sensitive. By default, fd uses 'smart case' for queries.\
+            This option disables smart case.");
+    doc!(h, "absolute-path"
+        , "Show absolute instead of relative paths"
+        , "Shows the result of a pattern match as an absolute path instead of relative path.");
+    doc!(h, "follow"
+        , "Follow symbolic links"
+        , "Includes a symbolic links destination in the pattern match.");
+    doc!(h, "full-path"
+        , "Search full path (default: file-/dirname only)"
+        , "Searches the full path of directory. By default, fd only navigates file-/dirname.");
+    doc!(h, "null_separator"
+        , "Separate results by the null character"
+        , "Separates each pattern match result by the null character.");
+    doc!(h, "depth"
+        , "Set maximum search depth (default: none)"
+        , "Set the limit of the maximum search depth in a pattern match query. By default, \
+            none is the max-depth.");
+    doc!(h, "file-type"
+        , "Filter by type: f(ile), d(irectory), s(ymlink)"
+        , "Filter the search by type:\n\
+            f file          for file\n\
+            d directory     for directory\n\
+            s symlink       for symbolic links.");
+    doc!(h, "extension"
+        , "Filter by file extension"
+        , "Filter the search to only match for a filetype extension");
+    doc!(h, "color"
+        , "When to use color in the output:\n\
+            never, auto, always (default: auto)"
+        , "Declare when to use color for the pattern match output:\n\
+            auto (default)\n\
+            never\n\
+            always");
+    doc!(h, "threads"
+        , "Set number of threads to use for searching:\n\
+            (default: number of available CPU cores)");
+    doc!(h, "max-buffer-time"
+        , "the time (in ms) to buffer, before streaming to the console"
+        , "Amount of time in milliseconds to buffer, before streaming to the console the results");
+    doc!(h, "pattern"
+        , "the search pattern, a regular expression (optional)"
+        , "The search pattern to use for a match. By default, uses string match. Can use a \
```

Comment:
Actually, the pattern is always treated as a regular expression. I don't think we need an extended help text here.